src/number.cr

```diff
@@ -240,6 +240,16 @@ struct Number
     self
   end
 
+  # Returns `true` if value is equal to zero.
+  #
+  # ```
+  # 0.zero? #=> true
+  # 5.zero? #=> false
+  # ```
+  def zero? : Bool
+    self == typeof(self).zero
```

Comment:
Read @MakeNowJust's comment. This is a more general approach, comparing directly to the type's definition of "zero"